ikarus/assembler/simpleAssemblers.inl

```diff
@@ -68,23 +79,42 @@ namespace Ikarus {
   }
 
   template <typename Basis, typename FEContainer>
-  Eigen::SparseMatrix<double> &SparseFlatAssembler<Basis, FEContainer>::getMatrixImpl(
-      const FERequirementType &fErequirements) {
-    if (!isOccupationPatternCreated) createOccupationPattern();
-    if (!arelinearDofsPerElementCreated) createlinearDofsPerElement();
-    spMat.coeffs().setZero();
+  void SparseFlatAssembler<Basis, FEContainer>::assembleRawMatrixImpl(const FERequirementType &feRequirements,
+                                                                      Eigen::SparseMatrix<double> &assemblyMat) {
+    assemblyMat.coeffs().setZero();
     Eigen::MatrixXd A;
     for (size_t elementIndex = 0; const auto &fe : this->finiteElements()) {
       A.setZero(fe.size(), fe.size());
-      fe.calculateMatrix(fErequirements, A);
+      fe.calculateMatrix(feRequirements, A);
       assert(std::sqrt(elementLinearIndices[elementIndex].size()) == A.rows()
              && "The returned matrix has wrong rowSize!");
       assert(std::sqrt(elementLinearIndices[elementIndex].size()) == A.cols()
              && "The returned matrix has wrong colSize!");
       for (Eigen::Index linearIndex = 0; double matrixEntry : A.reshaped())
-        spMat.coeffs()(elementLinearIndices[elementIndex][linearIndex++]) += matrixEntry;
+        assemblyMat.coeffs()(elementLinearIndices[elementIndex][linearIndex++]) += matrixEntry;
       ++elementIndex;
     }
+  }
+
+  template <typename Basis, typename FEContainer>
+  Eigen::SparseMatrix<double> &SparseFlatAssembler<Basis, FEContainer>::getRawMatrixImpl(
+      const FERequirementType &feRequirements) {
+    std::call_once(sparsePreProcessorRaw, [&](){
+      createOccupationPattern(spMatRaw);
+      createLinearDOFsPerElement(spMatRaw);
+    });
+    assembleRawMatrixImpl(feRequirements, spMatRaw);
+    return spMatRaw;
+  }
+
+  template <typename Basis, typename FEContainer>
+  Eigen::SparseMatrix<double> &SparseFlatAssembler<Basis, FEContainer>::getMatrixImpl(
+      const FERequirementType &feRequirements) {
+    std::call_once(sparsePreProcessor, [&](){

```

Comment:
The call once statements can be moved to `assembleRawMatrixImpl` to reduce code duplication, don't they?